kolibri/plugins/pdf_viewer/assets/src/views/PdfRendererIndex.vue

```diff
@@ -84,30 +84,40 @@
           :layout8="{ span: showSideBar ? 6 : 8 }"
           :layout12="{ span: showSideBar ? 9 : 12 }"
         >
-          <RecycleList
+          <DynamicScroller
             id="pdf-container"
             ref="recycleList"
             :items="pdfPages"
-            :itemHeight="itemHeight"
+            :minItemSize="itemHeight"
             :emitUpdate="true"
+            :buffer="1.5 * itemHeight"
             class="pdf-container scroller-height"
             keyField="index"
             @update="handleUpdate"
           >
             <template #default="{ item }">
-              <PdfPage
-                :key="item.index"
-                :pageNum="item.index + 1"
-                :pdfPage="pdfPages[item.index].page"
-                :pageReady="pdfPages[item.index].resolved"
-                :firstPageHeight="firstPageHeight || 0"
-                :firstPageWidth="firstPageWidth || 0"
-                :scale="scale || 1"
-                :totalPages="pdfPages.length"
-                :eventBus="eventBus"
-              />
+              <DynamicScrollerItem
+                :item="item"
+                :active="active"
+                :sizeDependencies="[
+                  item.height,
```

Comment:
Item.height is undefined here, as the items of the is are the `pdfPages` array, and each item has this structure:
`{
  page: null,
  resolved: false,
  index: i,
}`